.devcontainer/devcontainer.json

```
@@ -0,0 +1,67 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
+// README at: https://github.com/devcontainers/templates/tree/main/src/python
+{
+  "name": "Python 3",
+  "build": {
+    "dockerfile": "../docker/Dockerfile",
+    "context": "..",
+    "target": "base"
+  },
+
+  // Use 'forwardPorts' to make a list of ports inside the container available locally.
+  "forwardPorts": [8888],
+
+  // Use 'postCreateCommand' to run commands after the container is created.
+  // Populate the yarn cache
+  "postCreateCommand": "micromamba run pip install -e .",
```

Maybe with the `dev` extra?

```suggestion
  "postCreateCommand": "micromamba run pip install -e \".[dev]\"",
```

---

That step is needed only to build the JavaScript assets. All python dependencies are stored in a dedicated layer:

```
# Install all python dependencies only with pip to maximize using its cache
RUN --mount=type=cache,target=/jupyterlab/pip-cache,uid=${NEW_MAMBA_USER_ID},gid=${NEW_MAMBA_USER_GID} \
    PIP_CACHE_DIR="/jupyterlab/pip-cache" SKIP_JUPYTER_BUILDER=1 micromamba run python -m pip install -e  ".[dev,docs,test]" && \
    cd / && \
    rm -rf /home/$MAMBA_USER/jupyterlab_cache
```

So no need for the extra there.